src/gt4py/next/iterator/embedded.py

```diff
@@ -1662,6 +1662,13 @@ def impl(*args):
     return impl
 
 
+@builtins.cond.register(EMBEDDED)
+def cond(cond_, t, f):
+    if hasattr(cond_, "as_scalar"):
```

Comment:
As discussed: In case this is for 0d fields something along the lines of 
```suggestion
    if isinstance(cond_, Field) and cond_.ndim == 0:
```
would greatly improve readability. Otherwise a comment explaining this special casing would be nice.

Reply:
I copied this code from previous cycle's integration branch. @havogt Can you confirm that this check was meant for 0-dim fields?